### DISCUSSION TITLE
allow to run it in a subdirectory with reverse proxy

### DISCUSSION THREAD

**Discussion by tramir** [nextcloud/all-in-one#839] [Ideas] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Set up reverse proxy following [the guidelines for nginx](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md), with the only difference that the location is `/nextcloud` instead of `/`
2. Create the AIO container with the suggested settings (`APACHE_PORT=11000`)
3. Start the AIO container. Type in the domain that will be used for Nextcloud.

### Expected behavior 
The setup continues and the nextcloud containers are created.

### Actual behavior 
Error: `Domain does not point to this server or the reverse proxy is not configured correctly. See the mastercontainer logs for more details. ('sudo docker logs -f nextcloud-aio-mastercontainer')`

### Host OS 
Synology DSM 7.1

#### Nextcloud AIO version 
Nextcloud AIO v1.4.1

#### Current channel 
N/A

#### Other valuable info 
I suppose this behavior is to be expected, because the domain itself points to something else. Only the subdirectory would point to the nextcloud instance. But the setup process doesn't allow me to enter a domain/subdirectory (error: `Domain is not a valid domain!`). The setup continues if I change my reverse proxy setup to point `/` to the nextcloud instance. But I'd really like to be able to use a subdomain. How should I proceed?

**Comment by szaimen** [comment]
This is a feature request

**Comment by szaimen** [comment, marked as answer]
The easiest workaround for now would be pointing another subdomain onto your home network...

**Reply by szaimen** [reply]
another possibility is this: https://github.com/nextcloud/all-in-one/tree/main/manual-install

**Reply by tramir** [reply]
Unfortunately, I don't think that's possible. I have only one domain (no subdomains) and a bunch of services running in subdirectories.

**Reply by tramir** [reply]
Would it be possible to proceed as follows:
1. Set up the reverse proxy to forward `/` to nextcloud
2. Proceed with the installation
3. Modify config.php to change the root folder?

**Reply by szaimen** [reply]
not that I am aware of

**Reply by szaimen** [reply]
I still think using a subdomain is the easiest way of doing this. There are free ones available, too. Simply set a CNAME entry for the subdomain so that it points to your usual domain and you are done. (of course the reverse proxy needs to be configured correctly, too.)

**Reply by szaimen** [reply]
E.g. here is one free provider: https://desec.io/

**Reply by tramir** [reply]
OK, in the end I decided to go the new subdomain route. I got a new subdomain, configured the reverse proxy accordingly. I recreated the AIO container. I can confirm that I get a 48-character string when I access the new subdomain, which I suppose means that the reverse proxy is set up ok. But I still get the same error: `Domain does not point to this server or the reverse proxy is not configured correctly. See the mastercontainer logs for more details. ('sudo docker logs -f nextcloud-aio-mastercontainer')`

Checking the logs I see this:
`2022-06-23T21:41:15.392560358Z [Thu Jun 23 21:41:15.392225 2022] [php:notice] [pid 127] [client 127.0.0.1:59840] The response of the connection attempt to "XXXXX:443" was: , referer: https://192.168.147.200:8080/containers`

I can confirm that I get the 48-character hash if I access XXXXX:443, so I'm really stumped why this is happening. There's no other service using port 443 on this subdomain (there are on other subdomains, but the reverse proxy should take care of that). I googled and couldn't find any indication of what else I should do to get this over the hump. So any suggestion is much appreciated...

**Reply by szaimen** [reply]
okay great! This looks good in general! However it seems like the mastercontainer is not able to connect to the configured subdomain to get the correct value. Maybe for a test try to restart the mastercontainer and see if it works now?

**Reply by tramir** [reply]
Thanks for the quick reply! I restarted the container, but still have the same error. One thing I noticed is that, if I try to `wget` the same address that aio-domaincheck seems to be using (`https://XXXXX:443`), I get a certificate error unless I use `--no-check-certificate`. Does aio-domaincheck require a valid certificate?

**Reply by szaimen** [reply]
> Does aio-domaincheck require a valid certificate?

yes. Don't you have one?

**Reply by tramir** [reply]
I didn't. I do now :) And I'm able to move on. Maybe this should be added to the documentation? I know it's easy to get a Letsencrypt cert, but it wasn't obvious to me that it was needed. But maybe it's just because it's late here :)

In any case, many many thanks for all the help -- and for a great idea and product!

**Reply by szaimen** [reply]
I'll try to improve the logging with https://github.com/nextcloud/all-in-one/pull/859

**Comment by szaimen** [comment]
Another possibility is this: https://github.com/nextcloud/all-in-one/tree/main/manual-install

**Reply by tramir** [reply]
Thanks for suggesting this. I took a look at the manual install and I'm not sure what I should tinker with to get this done. It seems the process still requires a domain (`$NC_DOMAIN`) with no possibility that I can see for a subdirectory. At least that's what the Caddyfile and the test in start.sh seem to do. Would you have any guidance?

**Reply by tramir** [reply]
I'd really like to get this working because I think AIO is a super great idea. And I had nextcloud running in a subdirectory until now on a debian machine with the same reverse-proxy setup as the one I'm trying now. I have no experience with caddy though...

**Reply by ManOki** [reply]
see #1672 and [this](https://gist.github.com/ManOki/640a6bad8b4b199191733f2b2338b2ab)
